csrc/codegen.cpp

```diff
@@ -1281,6 +1281,108 @@ class CudaKernelGenerator : private kir::ConstIrVisitor {
     }
   }
 
+  void handle(const ArgsortOp* aop) final {
+    NVF_ERROR(isAligned(), "Unaligned argsort not supported");
+
+    const auto& parallel_dimension_map =
+        kernel_->summary().parallel_dimension_map;
+
+    NVF_ERROR(aop->out()->isA<kir::TensorIndex>());
+    const auto output = aop->out()->as<kir::TensorIndex>();
+
+    auto sorted_logical_id = output->view()->getLogicalDomain().at(aop->dim());
+    auto sorted_ids = DependencyCheck::getAllValsBetween(
```

Comment:
I know you said explicitly that we don't care much about safety check.

But for my own curiosity, our assumption here is that `sorted_ids` only contains sorted IDs, but in reality, we might pick up non-sorted IDs if they get merged with the sorted ID in transformations?